### ISSUE TITLE
`inquirer@10.1.6` has a missing type for `message` in prompt function

### ISSUE THREAD

**Issue by mikeseese** [SBoudrias/Inquirer.js#1495]
I have this code:

``` js
const response = await inquirer.prompt({
  type: "confirm",
  name: "continue",
  message: `\twhats up?`,
  default: false,
});
```

but the types included in `inquirer@10.1.6` provide an error for `message` (I was sure to uninstall `@types/inquirer` during my upgrade from v9):

![image](https://github.com/user-attachments/assets/e84cb791-568b-4000-9100-07917825b133)

It seems that `message` is missing from `QuestionMap`:

https://github.com/SBoudrias/Inquirer.js/blob/2cc9e311a38b8074fd1b30c073f3a7a506c8c75f/packages/inquirer/src/types.mts#L29-L41

but the missing type is being referenced by `PromptConfigMap`:

https://github.com/SBoudrias/Inquirer.js/blob/2cc9e311a38b8074fd1b30c073f3a7a506c8c75f/packages/inquirer/src/types.mts#L52-L57

I confirmed that the logic works if I ignore the type:

``` bash
yarn run v1.22.22
$ ts-node packages/deployment/src/docker/script.ts template-match-staging
?       whats up? (y/N
```

**Comment by SBoudrias** [comment]
Does it work with an array? `inquirer.prompt([{ ... }])`?

It's weird because there's multiple tests covering those... Which version of TS are you using? Did you try updating it?

**Comment by mikeseese** [comment]
Wrapping it in an array does not work:

![image](https://github.com/user-attachments/assets/c20fcb51-ef00-4d79-96de-e6d97c7c04e1)

I'm using `typescript@5.4.5`; Updating to `5.5.4` doesn't fix it. I can also confirm that the error is prompted in VSCode, ts-node, and tsc.

The code works if the type is ignored or if the tests are using a transpile only flag (and not type checking), so perhaps that could be why the tests are passing (I haven't looked at them myself)?

Note, my original analysis of how `QuestionMap` was missing `message` was just wrong; I misunderstood how the typing is working, but it makes sense now.

Changing `QuestionMap` to this seems to resolve the issue:

```
export interface QuestionMap {
    input: typeof input;
    select: typeof select;
    /** @deprecated `list` is now named `select` */
    list: typeof select;
    number: typeof number;
    confirm: typeof confirm;
    rawlist: typeof rawlist;
    expand: typeof expand;
    checkbox: typeof checkbox;
    password: typeof password;
    editor: typeof editor;
}
```

**Comment by SBoudrias** [comment]
Not a direct solution to your issue that I didn't have time to investigate (also, weirdly I've been able to migrate another TS codebase successfully to v10 🤷🏻)

But, did you consider switching to `@inquirer/prompts`? It's a much simpler interface, and has good strong typings. `inquirer` interfaces are old and very complex to type properly due to the old-style node patterns it relies upon.

**Comment by mikeseese** [comment]
I thought about it for about 5 seconds before I saw "rewrite" and noped out haha. I'll give it a shot later today hopefully

**Comment by mikeseese** [comment]
I moved to `@inquirer/prompts` and everything is working (understandably due to the much simpler types/usage. I even prefer the interface in `@inquirer/prompts`. Thanks for the gentle nudge! Up to you if you want to close this as a wont-fix or stating the solution is to upgrade/downgrade 🤷

**Comment by SBoudrias** [comment]
We released new core types today with `inquirer@10.2.0`. Would appreciate people trying it out and bringing feedback - cheers!
